File: src/main/java/info/dddeurope/lab/app/repositories/PostRepository.java
package info.dddeurope.lab.app.repositories;

import java.util.HashMap;
import java.util.Map;

import info.dddeurope.lab.app.domain.PostAggregate.Post;

/*

    Here we use a simple in-meomory key-map value to represent a document-based database.
    In real-world, you would instead use a real database, such as MongoDB.

*/

public class PostRepository {

    Map<String, Post> posts;
    
    public PostRepository() {
        this.posts = new HashMap<>();
    }

    void set(String postId, Post post) {
        this.posts.put(postId, post);
    }

    Post get(String postId) {
        return this.posts.get(postId);
    }

}

File: DotNet/App/Repositories/PostRepository.cs
using System.Collections.Generic;
using App.Domain.PostAggregate;

namespace App.Repositories
{
    /*
        Here we use a simple in-memory key-map value to represent a document-based database.
        In real-world, you would instead use a real database, such as MongoDB.
    */

    public class PostRepository
    {
        readonly Dictionary<string, Post> _posts;

        public PostRepository()
        {
            _posts = new Dictionary<string, Post>();
        }

        public void Set(string postId, Post post)
        {
            _posts[postId] = post;
        }

        Post Get(string postId)
        {
            return _posts.TryGetValue(postId, out var value) ? value : null;
        }
    }
}